src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py

```diff
@@ -106,9 +106,10 @@ def create_firewall_rule(db_context, cmd, resource_group_name, server_name, star
     #    firewall_client.create_or_update(resource_group_name, server_name, firewall_name , start_ip, end_ip),
     #    cmd.cli_ctx, '{} Firewall Rule Create/Update'.format(logging_name))
 
-    firewall = firewall_client.create_or_update(resource_group_name, server_name, firewall_name, start_ip,
-                                                end_ip).result()
-    return firewall.name
+    firewall = firewall_rule_create_func(firewall_client, resource_group_name, server_name, firewall_rule_name=firewall_name,
+                                         start_ip_address=start_ip, end_ip_address=end_ip)
+
+    return firewall.result()
```

Comment:
Should we return `firewall.result().name` to keep align with before?

Reply:
good catch,,, thanks..!